client/gutenberg/extensions/map/component.js

```diff
@@ -0,0 +1,338 @@
+							<p>{ caption }</p>
+						</Fragment>
+					) }
+			</InfoWindow>
+		);
+		return (
+			<Fragment>
+				<div className="wp-block-jetpack-map__gm-container" ref={ this.mapRef }>
+					{ mapMarkers }
+				</div>
+				{ infoWindow }
+				{ addPoint }
+			</Fragment>
+		);
+	}
+	componentDidMount() {
+		const { api_key } = this.props;
+		if ( api_key ) {
+			this.loadMapLibraries();
+		}
+	}
+	componentWillUnmount() {
+		this.debouncedSizeMap.cancel();
+	}
+	componentDidUpdate( prevProps ) {
+		const { api_key, children, points, map_style, map_details } = this.props;
+		const { map } = this.state;
+		if ( api_key && api_key.length > 0 && api_key !== prevProps.api_key ) {
+			window.mapboxgl = null;
+			this.loadMapLibraries();
+		}
+		// If the user has just clicked to show the Add Point component, hide info window.
+		// AddPoint is the only possible child.
+		if ( children !== prevProps.children && children !== false ) {
+			this.clearCurrentMarker();
+		}
+		if ( points !== prevProps.points ) {
+			this.setBoundsByMarkers();
+		}
+		if ( points.length !== prevProps.points.length ) {
+			this.clearCurrentMarker();
+		}
+		if ( map_style !== prevProps.map_style || map_details !== prevProps.map_details ) {
+			map.setStyle( this.getMapStyle() );
+		}
+	}
+	/* Event handling */
+	onMarkerClick = marker => {
+		const { onMarkerClick } = this.props;
+		this.setState( { activeMarker: marker } );
+		onMarkerClick();
+	};
+	onMapClick = () => {
+		this.setState( { activeMarker: null } );
+	};
+	clearCurrentMarker = () => {
+		this.setState( { activeMarker: null } );
+	};
+	updateActiveMarker = updates => {
+		const { points } = this.props;
+		const { activeMarker } = this.state;
+		const { index } = activeMarker.props;
+		const newPoints = points.slice( 0 );
+
+		assign( newPoints[ index ], updates );
+		this.props.onSetPoints( newPoints );
+	};
+	deleteActiveMarker = () => {
+		const { points } = this.props;
+		const { activeMarker } = this.state;
+		const { index } = activeMarker.props;
+		const newPoints = points.slice( 0 );
+
+		newPoints.splice( index, 1 );
+		this.props.onSetPoints( newPoints );
+		this.setState( { activeMarker: null } );
+	};
+	// Various map functions
+	sizeMap = () => {
+		const { map } = this.state;
+		const mapEl = this.mapRef.current;
+		const blockWidth = mapEl.offsetWidth;
+		const maxHeight = window.innerHeight * 0.8;
+		const blockHeight = Math.min( blockWidth * ( 3 / 4 ), maxHeight );
+		mapEl.style.height = blockHeight + 'px';
+		map.resize();
+		this.setBoundsByMarkers();
+	};
+	setBoundsByMarkers = () => {
+		const { zoom, points, onSetZoom } = this.props;
+		const { map, activeMarker, mapboxgl, zoomControl, fit_to_bounds } = this.state;
+
+		const bounds = new mapboxgl.LngLatBounds();
+		if ( ! map || ! points.length || activeMarker ) {
+			return;
+		}
+		points.forEach( point => {
+			bounds.extend( [ point.coordinates.longitude, point.coordinates.latitude ] );
+		} );
+		map.setCenter( bounds.getCenter() );
```

Comment:
Should this be as an `else` clause after next `if`, since in it we then call `map.fitBounds` which basically does what this `setCenter` already did, just it also resets the zoom, orientation and adds some padding.

Reply:
Huh, yeah, I was under the impression you did need `setCenter` in addition to `fitBounds`, but clearly this doesn't make much sense. I moved it after the `if` block (no `else` clause necessary since the block contains a `return). Done here: https://github.com/Automattic/wp-calypso/pull/28498/commits/f473f897b52282b21e5b8bfe3ec776a0592521f6